resources/store/state.js

```diff
@@ -15,18 +15,11 @@ module.exports = {
 
 	unreviewedCount: 0,
 
-	/**
-	 * @TODO Currently four possible status states exist:
-	 * - "success"
-	 * - "error"
-	 * - "pending"
-	 * - null (default)
-	 *
-	 * It may be time to think about enforcing a consistent and constrained
-	 * list of states here and in the corresponding mutation, maybe by defining
-	 * publish states in an external object and importing them here
-	 */
-	publishStatus: null,
-
-	userStats: {}
+	publishPending: false,
+
+	userStats: {},
+
+	toastNotifications: [],
+
+	cardStackMessage: null
```

Comment:
I would consider using more descriptive names for `toastNotifications` and `cardStackMessage` to give a sense of which kinds of messages get what treatment. The UI element used for a message is a presentational detail that we should really only care about at the level of components. Inside the Vuex store I think it would make more sense to name things based on functionality.

Would it be correct to say that toast notifications are always concerned with publishing images, while cardstack messages are always concerned with fetching them? 

Alternatively, if toast messages are always about a *specific* image while the cardstack messages are always about an entire *queue* of images, maybe that would be a meaningful way to differentiate. `imageNotifications` vs `queueMessage` maybe?

Reply:
I changed `toastNotifications` to `imageMessages`, which I agree makes more sense. I started consolidating all of the queue messages into state and the code became way more complex and verbose. Plus, since one of the queue messages was a standard message and the other 3 were our custom UserMessage component, I was going to have to use the UserMessage component for the fetch error, for which a standard error message is more appropriate.

I decided that this change was overcomplicated and made the code harder to read, so I revert it and removed cardStackMessage from state altogether. Instead, I added `fetchError`, which is a boolean value for each queue indicating whether or not the fetch failure message should display.